### ISSUE TITLE
"Begin a new dialog" action doesn't work with property expressions

### ISSUE THREAD

**Issue by Brankec** [microsoft/BotFramework-Composer#9050]
## Describe the bug
In Bot Framework Composer there is an action "Begin a new dialog". This Action has the option for "Expression" input, however giving it a property with a dialog name value does not work. I am aware the cause for this is because of the DialogSet not being registered but I will explain that further in Additional Context section.

## Version
Composer Version 2.1.2
SDK version 4.15.0

## Browser

- [X] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [ ] macOS
- [X] Windows
- [ ] Ubuntu

## To Reproduce

1. Create a new C# Echo bot in the Composer.
2. Create a new dialog to redirect to using "Begin a new dialog" action.
3. In the main dialog (original dialog) create a new property using "Set a property" action, for testing purposes I named my property "dialog.ThisIsATestProperty" and assigned it the string value of the Dialog name I created in **Step 2. of this list**.
4.  Use that property you just created and assign it to "Begin a new dialog" action in the input field "Dialog name" as an expression.
5. Start the bot and have it attempt to redirect you to the new dialog you created.
6. The bot throws an error `Object reference not set to an instance of an object.` 

## Expected behavior

Begins a new dialog with the value contained inside the property assigned to it as an expression

## Screenshots

This is my dialog tree
![image](https://user-images.githubusercontent.com/33580274/158337695-a854c354-d036-4807-a74b-9b26749ee5ea.png)

This is the property
![image](https://user-images.githubusercontent.com/33580274/158337947-027d2fb8-1ff0-489c-b92c-6757191adbe2.png)

This is the dialog I assigned a property to
![image](https://user-images.githubusercontent.com/33580274/158337914-6dddf2a9-b3aa-4b8e-a418-a70d4f7968f9.png)

## Additional context

I have found a similar [closed issue](https://github.com/microsoft/BotFramework-Composer/issues/8821) where the fix for it was to either:
1. Predefine all the dialogs by spamming "Begin a new dialog" in another trigger so that the DialogSet gets loaded
2. Add a property "dialogs" inside the root of your Dialog and add an array of all the dialogs so that they can be loaded into the DialogSet

Option 1. Is not feasible as I have attempted this before but because of the number of Dialogs I would have to predefine would choke the bot during loading times and any reqeust to it times out by the time it finishes loading, this is also not a very clean solution.

Option 2. I have tried however it keeps saying `AssecoChatbot.dialog error: Value cannot be null. (Parameter ‘dialog’)`

I am out of alternatives and I've been stuck on this issue for the past week. Any help would be incredibly appreciated.

**Comment by Brankec** [comment]
In Additional context I mentioned that option 2. does not work. I found out what the problem is however I am not sure how to fix it.

The problem is because my Root Dialog has a "Begin a new dialog" to the dialog I added inside the property "dialogs", more specifically one of the children from my Root Dialog has a "begin a new dialog" that calls the dialog I defined inside the property "dialogs". Is there a way to circumvent this issue? I am not sure why this is even happening... If it's complaining that I have duplicate dialogs in my "dialogs" property then why doesn't it work without it?

**Comment by Brankec** [comment]
I found a workaround. It seems like the issue only stems if my children have any of the "dialogs" in their "begin a new dialog", however the same does not apply if your parents have that. Simple solution was to create a new dialog that has no other children and has defined dialogSet in "dialogs" property. Then I simply call that dialog and do all of my redirecting to other dialogs there.

There is one small problem which is very long loading times, when starting the bot for the first time, if you have like 10+ dialogs defined but that should probably be another issue post. I think I milked this one enough and is starting to get off topic.

Also these mentions of dialog is getting really confusing to read/comprehend even for me, so sorry about that too.

**Comment by stevkan** [comment]
@Brankec, if I understand your problem correctly, then I believe the following steps should help you.

First, change your property from being assigned to the 'dialog' state bucket to the 'conversation' state bucket. Properties in the 'dialog' bucket only exist for the duration that the dialog they are assigned to is running. So, as your bot flows from dialog to dialog, anything properties in dialog "A" will be lost as the bot moves to dialog "B".

`conversation.TestDialogName`

As for setting a property and then accessing it in the same "main" dialog, I first create a "SingleUseCancelDialog" dialog. This shouldn't be confused with the "cancel" dialog used for user interruption (tho it is doing the same thing, but without all the conversational text). This "SingleUseCancelDialog" dialog should only consist of the "Cancel all dialogs" action.

![image](https://user-images.githubusercontent.com/18403946/158466922-a0b3763d-dae6-40bb-9cad-91947a9dab85.png)

In the main dialog, I have set the property at the top that has a text value matching the dialog I wish to have called.

I then include the "Replace this dialog" action in the loop. This will only run once when the user joins. In this action, I call the "SingleUseCancelDialog" dialog I defined earlier. With no interruption or delay to the user, this allows the properties to be registered and available moving forward.

At the end of the main dialog, I add the "Begin a new dialog" action which takes the `conversation.TestDialogName` value and calls the dialog of the same name. In my case, the dialog, "SendAdaptiveCard", is called which will send a simple adaptive card when reached.

![image](https://user-images.githubusercontent.com/18403946/158467292-d6e10542-9221-422c-b739-2263b238f475.png)

![dialogFromPropertyName](https://user-images.githubusercontent.com/18403946/158470744-87dabbcf-113e-4636-8566-96eda3e2e041.gif)

**Comment by Brankec** [comment]
@stevkan actually my issue is a little diffrent if we are talking about this comment of mine:
> In Additional context I mentioned that option 2. does not work. I found out what the problem is however I am not sure how to fix it.
> 
> The problem is because my Root Dialog has a "Begin a new dialog" to the dialog I added inside the property "dialogs", more specifically one of the children from my Root Dialog has a "begin a new dialog" that calls the dialog I defined inside the property "dialogs". Is there a way to circumvent this issue? I am not sure why this is even happening... If it's complaining that I have duplicate dialogs in my "dialogs" property then why doesn't it work without it?

But it is to be expected since Im throwing around the word "dialog" with 3 different meanings so it gets confusing when I'm talking about the "Begin a new dialog" action, the "dialogs" property inside the json root of the main AdaptiveDialog and "dialog" as a singular dialog name I wish to redirect my self to.

I am giving my self a brain fog just by reading what I said so sorry about that.

As for what my issue was. I figured out how to use expression input for my "Begin a new dialog" action, by simply defining "dialogs" inside the json root
![image](https://user-images.githubusercontent.com/33580274/158474642-d65cb189-875e-4285-9241-d860beb0a9b1.png)

But the issue was, if you already have "Begin a new dialog" that has the same dialog defined in that list above it will not work. It will throw an error `AssecoChatbot.dialog error: Value cannot be null. (Parameter ‘dialog’)`.
![image](https://user-images.githubusercontent.com/33580274/158475595-6848965f-e4a6-4f8f-952e-7211b73a2de6.png)

I am aware that in the background "Begin a new dialog" adds the dialog inside the DialogSet so I suppose doing it manually once and then having "Begin a new dialog" do it the second time for the same dialog name makes the Bot Framework very 😠.

I have avoided this issue with a pretty simple fix I commented just before you. I now just have an issue with slow loading times because I have about 30 dialogs defined in "dialogs" property. I feel like that can't really be fixed so I will have to deal with that.
